agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -483,30 +565,46 @@ impl DeriveSchema for Stage {
         }
 
         match self {
-            Stage::AddFields(_) => todo!(),
+            Stage::AddFields(a) => add_fields_derive_schema(a, state),
             Stage::AtlasSearchStage(_) => todo!(),
             Stage::Bucket(_) => todo!(),
             Stage::BucketAuto(_) => todo!(),
-            Stage::Collection(_) => todo!(),
-            Stage::Count(_) => todo!(),
+            Stage::Collection(c) => {
+                let ns = Namespace(c.db.clone(), c.collection.clone());
+                state.result_set_schema = state
+                    .catalog
+                    .get(&ns)
+                    .ok_or(Error::UnknownReference(ns.to_string()))?
+                    .clone();
+                Ok(state.result_set_schema.to_owned())
+            }
+            Stage::Count(c) => {
+                state.result_set_schema = Schema::Document(Document {
+                    keys: map! {
+                        c.clone() => Schema::AnyOf(set!{Schema::Atomic(Atomic::Integer), Schema::Atomic(Atomic::Long), Schema::Atomic(Atomic::Double)})
```

Comment:
is this right, count can return a double?

Reply:
https://www.mongodb.com/docs/manual/reference/operator/aggregation/count/#behavior

Yes it can!

Reply:
It seems like it can, but not for realistic collections, I'm going to back off on this even though it technically can return double if we overflow 2^63 documents.